### ISSUE TITLE
Google Closure Compiler is importing my extern functions

### ISSUE THREAD

**Issue by JohannBlake** [google/closure-compiler#2603]
I have created an extern in a javascript file and specified it as part of the Google Closure Compiler (GCC) command line option. I am compiling with advanced mode. GCC is taking the function in my extern and placing it in the compiled code. I have no idea why it would do this. GCC is suppose to recognize that the extern function is in a separate file. When I export the object, it will rename the object and leave the object's function names alone BUT it will create a copy of the entire extern function in the compiled code.

I have tried many variations (too numerous here to list) to see how to prevent GCC from doing this but nothing has worked.

My extern:

    var MyCustomResizer = {
        "onResize": function (a, b) {
        },
        "detach": function () {
        }
    }

I exported the object as follows:

    window["MyCustomResizer"] = MyCustomResizer;

My app using the "detach" function:

    MyCustomResizer.detach();

My compiler settings:

    java -jar closure-compiler/compiler.jar \
        --compilation_level ADVANCED_OPTIMIZATIONS \
        --externs   scripts/externs/resizer-extern.js \
        --js_output_file scripts/release/myapp.js \
        --warning_level VERBOSE \
        --language_out ECMASCRIPT5 \
        --language_in=ECMASCRIPT_2017 \
        --js scripts/base.js

And the generated compiled output contains this:

    ha.detach();
    ...
    var ha = {
        onResize: function () {
        }, detach: function () {
        }
    };

**Comment by JohannBlake** [comment]
It turns out that when you specify extern files, you MUST use the --extern option in front of every extern file. I only had it on the first one:

Incorrect:

    java -jar closure-compiler/compiler.jar \
        --compilation_level ADVANCED_OPTIMIZATIONS \
        --externs scripts/externs/jQuery/jquery-1.9-externs.js  \
                  scripts/externs/third-party.js \
        --js_output_file scripts/release/servetus-min.js \


Correct:

    java -jar closure-compiler/compiler.jar \
        --compilation_level ADVANCED_OPTIMIZATIONS \
        --externs scripts/externs/jQuery/jquery-1.9-externs.js  \
        --externs scripts/externs/third-party.js \
        --js_output_file scripts/release/servetus-min.js \

I find it very strange that the compiler just ignores the one without the --externs but just goes ahead anyway and copies its functions into the compiled code. This should not be allowed and a warning should be issued. This took an entire day to track down. 

This should be addressed in an upgraded version or at the very minimum documented.
